packages/@ember/-internals/glimmer/lib/component.ts

```diff
@@ -727,13 +728,10 @@ const Component = CoreView.extend(
       this[ROOT_REF] = new RootReference(this);
       this[BOUNDS] = null;
 
-      // If in a tagless component, assert that no event handlers are defined
-      assert(
-        // tslint:disable-next-line:max-line-length
-        `You can not define a function that handles DOM events in the \`${this}\` tagless component since it doesn't have any DOM element.`,
-        this.tagName !== '' ||
-          !this.renderer._destinedForDOM ||
-          !(() => {
+      if (DEBUG) {
+        let eventName = '';
+        if (this.tagName === '') {
+          eventName = (() => {
```

Comment:
Can we make this an array instead of a single eventName string? In theory it is possible to have more than one, and I'd love to list them all instead of playing "whack-a-mole"...

Reply:
It's an array now.